src/xmpp_ofc_simple_ids_switch.erl

```diff
@@ -0,0 +1,227 @@
+-define(FLOW_PRIORITY, 200).
+-define(FLOW_DROP_PRIORITY, 300).
+-define(COOKIE, <<0,0,0,0,0,0,0,200>>).
+-define(COOKIE_MASK, <<0,0,0,0,0,0,0,0>>).
+-define(FM_TIMEOUT_S(Type), case Type of
+                                idle ->
+                                    30;
+                                hard ->
+                                    120
+                            end).
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+
+-spec start_link(binary()) -> {ok, pid()} | ignore | {error, term()}.
+start_link(DatapathId) ->
+    {ok, Pid} = gen_server:start_link(?MODULE, [DatapathId], []),
+    {ok, Pid, subscriptions(), [init_flow_mod()]}.
+
+-spec stop(pid()) -> ok.
+stop(Pid) ->
+    gen_server:stop(Pid).
+
+-spec handle_message(pid(),
+                     {MsgType :: term(),
+                      Xid :: term(), 
+                      MsgBody :: [tuple()]},
+                     [ofp_message()]) -> [ofp_message()].
+handle_message(Pid, Msg, OFMessages) ->
+    gen_server:call(Pid, {handle_message, Msg, OFMessages}).
+
+
+%% ------------------------------------------------------------------
+%% gen_server Function Definitions
+%% ------------------------------------------------------------------
+
+init([DatapathId]) ->
+    %%Maybe this call should belong somewhere else...
+    schedule_flow_stats_request(DatapathId),
+
+    {ok, #state{datapath_id = DatapathId, fwd_table = #{}}}.
+
+handle_call({handle_message, {packet_in, _, MsgBody} = Msg, CurrOFMesssages},
+            _From, #state{datapath_id = Dpid, fwd_table = FwdTable0} = State) ->
+    case xmpp_ofc_util:packet_in_extract(reason, MsgBody) of
+        action ->
+            {OFMessages, FwdTable1} = handle_packet_in(Msg, Dpid, FwdTable0),
+            {reply, OFMessages ++ CurrOFMesssages,
+             State#state{fwd_table = FwdTable1}};
+        _ ->
+            {reply, CurrOFMesssages, State}
+    end;
+handle_call({handle_message, {flow_stats_reply, _, MsgBody} = _Msg,
+            CurrOFMessages}, _From,
+            #state{datapath_id = Dpid, fwd_table = FwdTable0} = State) ->
+    FlowEntries = proplists:get_value(flows, MsgBody),
+    case FlowEntries of
+        [_H|_T] ->
+            {OFMessages, FwdTable1, _} = lists:foldl(fun handle_flow_entry_stats/2,
+                                      {[], FwdTable0, Dpid}, FlowEntries),
+            {reply, OFMessages ++ CurrOFMessages,
+             State#state{fwd_table = FwdTable1}};
+        [] ->
+            {reply, CurrOFMessages, State}
+    end.
+
+handle_cast(_Request, State) ->
+    {noreply, State}.
+
+handle_info({send_flow_stats_request, DatapathId}, State) ->
+    lager:info("Called: handle_info()"),
+    send_flow_stats_request(DatapathId),
+    schedule_flow_stats_request(DatapathId),
+    {noreply, State};
+handle_info({remove_entry, Dpid, SrcMac},
```

Comment:
How are you getting this notification about entry being removed? I mean you're not subscribing for this type of message so I guess this callback is never called..